universal-login-sdk/lib/sdk.ts

```diff
@@ -127,18 +127,34 @@ class UniversalLoginSDK {
     return this.relayerApi.getConfig();
   }
 
+  async waitForStatus(messageId: string, timeout: number = 1000, tick: number = 20) {
```

Comment:
This function stinks as hell (too long, mixed abstraction layers), her I refactored it for you:

Util function:
```ts
async function retry(callback, timeout: number = 5000, tick: number = 1000) {
  let stopTime = now() + timeout;
  let result;
  do {
    ensure(stopTime <= now(), new Timeout)
    await sleep(tick);    
    result = callback;
  } while (!result);
  return result;
}
```

RelayerApi function:
```ts
async getMessageStatusIfMined() {
  const status =  await this.relayerApi.getStatusById(messageId);
  ensure(!status.error, new APIError(status))
  return status.transactionHash ? status : null; 
}
```

Sdk function:
```ts
async waitForStatus(messageId: string, timeout: number = 5000, tick: number = 1000) {
  return retry(getMessageStatusIfMined);
}
```

Add some nice error hierarchy in newly created `errors.ts`:
```ts
class SDKError extends Error {

}

class APIError extends SDKError {
  constructor(response) {
    super(response.error);
  }
}

class Timeout extends SDKError {

}
```

Reply:
#nextPR